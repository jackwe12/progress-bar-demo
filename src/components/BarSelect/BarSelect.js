import React from 'react';
import Dropdown from 'react-bootstrap/Dropdown';
import PropTypes from 'prop-types';
import './BarSelect.scss';

const DropButton =({current, setCurrent, bars})=>{

    return(
        <span >
            <Dropdown>
                <Dropdown.Toggle variant="success" id="dropdown-basic">  
                    # Progress{current + 1}               
                    {/* show current progress */}
                </Dropdown.Toggle>
           
                <Dropdown.Menu>
                    {/* when hit the progress button switch to the progress you choose */}
                    {bars.map((bar,index)=>   //takes the second parameter as index
                        <Dropdown.Item  onClick={()=>setCurrent(index)} key={index}> # Progress {index + 1}</Dropdown.Item> 
                    
                    )}
                </Dropdown.Menu>
                </Dropdown>
                


        </span>
    );
}

export default DropButton;

DropButton.propTypes = {
    setCurrent: PropTypes.func,
    bars: PropTypes.array,
    current: PropTypes.number,

}
