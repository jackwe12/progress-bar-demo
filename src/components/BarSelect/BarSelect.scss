@media screen and (max-width:500px){
.dropdown{
    width:100%;
}
}

@media screen and (max-width:350px){

    
}