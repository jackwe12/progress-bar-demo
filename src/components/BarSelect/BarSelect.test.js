import BarSelect from './BarSelect';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

configure({ adapter: new Adapter() });

const mockHandler = jest.fn();
const wrapper = mount(
    <BarSelect
        current = {0}
        setCurrent = {mockHandler}
        bars = {[10, 20, 30]}
    />);


describe('Testing <ControlBtn/>', ()=>{

    test('snapshot testing',()=>{
        const tree =renderer
            .create(     <BarSelect 
                        current = {0}
                        setCurrent = {mockHandler}
                        bars = {[10, 20, 30]}
            />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    });

    test('current progress is 1',()=>{
        expect(wrapper.find('#dropdown-basic').at(0).text()).toEqual('# Progress1')
    });



});
