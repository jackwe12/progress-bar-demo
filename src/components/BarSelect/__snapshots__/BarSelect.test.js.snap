// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <ControlBtn/> snapshot testing 1`] = `
<span>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-success"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      # Progress
      1
    </button>
  </div>
</span>
`;
